diesel/src/expression/expression_methods/global_expression_methods.rs

```diff
@@ -295,11 +295,10 @@ pub trait ExpressionMethods: Expression + Sized {
     ///     use self::posts::dsl::{posts, author_name};
     ///     let connection = establish_connection();
     ///
-    ///     let data = users.inner_join(posts)
+    ///     let data: QueryResult<Vec<String>> = users.inner_join(posts)
     ///         .filter(name.nullable().eq(author_name))
```

Comment:
Not specifically the `filter` line, just that this chain demonstrated the mixing of styles.

So maybe the guideline is to keep each line of the chain short and readable then
```Rust
let result = my_struct.foo_function()
  .bar_function().unwrap()
  .baz_with_longer_sig()
```